app/services/pii/attributes.rb

```diff
@@ -18,16 +18,45 @@ def self.new_from_encrypted(encrypted, user_access_key)
     end
 
     def self.new_from_json(pii_json)
-      attrs = new
-      return attrs unless pii_json.present?
+      return new unless pii_json.present?
       pii = JSON.parse(pii_json, symbolize_names: true)
-      pii.each_key { |attr| attrs[attr] = pii[attr] }
-      attrs
+      new_from_hash(pii)
+    end
+
+    def initialize(*args)
+      super
+      assign_all_members
     end
 
     def encrypted(user_access_key)
       encryptor = Pii::PasswordEncryptor.new
       encryptor.encrypt(to_json, user_access_key)
     end
+
+    def eql?(other)
+      to_json == other.to_json
+    end
+
+    def ==(other)
+      eql?(other)
+    end
+
+    def []=(key, value)
+      if value.is_a?(Hash)
+        super(key, Pii::Attribute.new(value))
+      elsif value.is_a?(Pii::Attribute)
+        super(key, value)
+      else
+        super(key, Pii::Attribute.new(raw: value))
+      end
+    end
+
+    private
+
+    def assign_all_members
+      self.class.members.each do |member|
+        self[member] = self[member]
```

Comment:
odd looking right? but that's the most elegant way I found of coercing all values to be `Pii::Attribute` objects.